### DISCUSSION TITLE
Kernel Source Code

### DISCUSSION THREAD

**Discussion by peyton-howe** [Joshua-Riek/ubuntu-rockchip#164] [General]
Are there any plans to make repo contain the linux kernel source itself rather than using various patches? I'm thinking something along the lines of how how raspberry pi has it setup with their linux [repo](url=https://github.com/raspberrypi/linux)?

Or is the source code for the kernel contained elsewhere and this one just contains the patches necessary to make everything run smoothly?

**Comment by Joshua-Riek** [comment]
I'm still figuring out how best to deal with the kernel source. My original intent was not to make any changes to the kernel. However, I have been put in a position where digging into the kernel and making my changes is necessary. I do agree that having multiple patches is not a very smooth process. I would like not to rebase anymore and commit directly, then merge any changes from Armbian as they come, and then tag the releases. This is something that I will work on but need time, I'm finishing up the next release, and there are a lot of kernel changes.

I have a fork of the kernel available below. But it's experimental, and I have been using it for testing purposes. The latest stable source is available through my ppa.

ppa: https://launchpad.net/~jjriek/+archive/ubuntu/rockchip/+packages
kernel fork: https://github.com/Joshua-Riek/linux-rockchip

**Comment by Joshua-Riek** [comment]
Just an update on this, the kernel is available at the repo below. No more patch files or rebasing!

https://github.com/Joshua-Riek/linux-rockchip
